src/integrations/blocks/structured-data-blocks.php

```diff
@@ -173,6 +173,75 @@ public function optimize_faq_images( $attributes, $content ) {
 		return $this->optimize_images( $attributes['questions'], 'answer', $content );
 	}
 
+	/**
+	 * Transforms the durations into a translated string containing the count, and either singular or plural unit.
+	 * For example (in en-US): If 'days' is 1, it returns "1 day". If 'days' is 2, it returns "2 days".
+	 * If a number value is 0, we don't output the string.
+	 *
+	 * @param number $days Number of days.
+	 * @param number $hours Number of hours.
+	 * @param number $minutes Number of minutes.
+	 * @return array Array of pluralized durations.
+	 */
+	private function transform_duration_to_string( $days, $hours, $minutes ) {
+		$strings = [];
+		if ( \intval( $days ) !== 0 ) {
+			$strings[] = \sprintf(
+			/* translators: %d expands to the number of day/days. */
+				\_n( '%d day', '%d days', $days, 'wordpress-seo' ),
+				$days
+			);
+		}
+		if ( \intval( $hours ) !== 0 ) {
+			$strings[] = \sprintf(
+			/* translators: %d expands to the number of hour/hours. */
+				\_n( '%d hour', '%d hours', $hours, 'wordpress-seo' ),
+				$hours
+			);
+		}
+		if ( \intval( $minutes ) !== 0 ) {
+			$strings[] = \sprintf(
+			/* translators: %d expands to the number of minute/minutes. */
+				\_n( '%d minute', '%d minutes', $minutes, 'wordpress-seo' ),
+				$minutes
+			);
+		}
+		return $strings;
+	}
+
+	/**
+	 * Formats the durations into a translated string.
+	 *
+	 * @param array $attributes The attributes.
+	 * @return string The formatted duration.
+	 */
+	private function build_duration_string( $attributes ) {
+		$days            = ( isset( $attributes['days'] ) ) ? ( $attributes['days'] ) : ( 0 );
+		$hours           = ( isset( $attributes['hours'] ) ) ? ( $attributes['hours'] ) : ( 0 );
+		$minutes         = ( isset( $attributes['minutes'] ) ) ? ( $attributes['minutes'] ) : ( 0 );
```

Comment:
For the reviewer: I use defensive coding here to access the attributes' keys only when they exist. Otherwise, a warning will be output. I feel like this is an ugly way of doing this, but I don't know how else to do this in PHP 🙈 
So, if you have a better and neater way of doing this, please leave a comment :)

Reply:
You can also:
```
$days    = $attributes['days'] ?? 0;
$hours   = $attributes['hours'] ?? 0;
$minutes = $attributes['minutes'] ?? 0;
```